src/labeler.ts

```diff
@@ -6,6 +6,7 @@ import {Minimatch, IMinimatch} from 'minimatch';
 interface MatchConfig {
   all?: string[];
   any?: string[];
+  branch?: string | string[];
```

Comment:
At the moment, the new field name in the yaml config file (`branch`) is not specific. We should make it more concrete so that there is no confusion and users can understand what type of branch is meant. How about `headBranch`? 
There is a [feature request](https://github.com/actions/labeler/issues/54) in the repository to add similar functionality, but for base branches. It would be great if you could add this in scope of your PR. I think we can reuse some logic that was introduced by you.